### PR TITLE
dsp: utils: Added macros to shift and convert float/fixed types

### PR DESCRIPTION
**Pull request by mrcamilletti**
The DSP library does not provide functions to shift and convert fixed representations in Q7, Q15, and Q31 formats to float or double and vice versa. This commit implements those macros using a shift argument to indicate the number of decimal places.

### CODE REVIEW

**Comment by mrcamilletti** [comment]
Hi @yperess, @stephanosio. Is there any possibility to review this?
Cheers!

File: include/zephyr/dsp/utils.h [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+ *
+ * There are separate functions for floating-point, Q7, Q15, and Q31 data types.
+ * @{
+ */
+
+/**
+ * @brief Convert a Q7 fixed-point value to a floating-point (float32_t) value with a left shift.
+ *
+ * @param src The input Q7 fixed-point value.
+ * @param n   The number of bits to left shift the input value (0 to 7).
```
Comment (yperess) [inline include/zephyr/dsp/utils.h:43]: Here and below, documentation is for `n` but arg is `m`.

**Review by yperess — CHANGES_REQUESTED** [review]
Minor nit on the documentation of the macros. But overall LGTM

File: include/zephyr/dsp/utils.h (hunk 1 above)
Comment (mrcamilletti) [reply include/zephyr/dsp/utils.h:43]: Thanks! I will fix this in the next push.

**Review by yperess — DISMISSED** [review]

File: tests/subsys/dsp/utils/prj.conf [hunk 2]
```diff
@@ -0,0 +1,10 @@
+CONFIG_ZTEST=y
+CONFIG_DSP=y
+CONFIG_DSP_BACKEND_CMSIS=y
+CONFIG_CMSIS_DSP=y
+CONFIG_CMSIS_DSP_SUPPORT=y
+CONFIG_REQUIRES_FULL_LIBC=y
+
+#To print floating point numbers, the full libc is required
+CONFIG_PICOLIBC=y
+CONFIG_REQUIRES_FLOAT_PRINTF=y
```
Comment (kartben) [inline tests/subsys/dsp/utils/prj.conf:10]: The comment is misplaced, but you only really need the below, AFAICT (just tested it, too)

```suggestion
CONFIG_REQUIRES_FLOAT_PRINTF=y
```

**Review by kartben — CHANGES_REQUESTED** [review]

File: tests/subsys/dsp/utils/prj.conf (hunk 2 above)
Comment (mrcamilletti) [reply tests/subsys/dsp/utils/prj.conf:10]: Nice catch! Thanks

**Review by kartben — APPROVED** [review]

**Review by yperess — APPROVED** [review]

**Pull request merged**